src/types/token_amount.ts

```diff
@@ -0,0 +1,58 @@
+// Utils
+import { BigNumber } from "../../utils/bignumber";
+
+import { Token } from "../types";
+
+export class TokenAmount {
+    public static fromAmount(symbol: string, amount: BigNumber): TokenAmount {
+        const tokenAmount = new TokenAmount(symbol);
+        tokenAmount.amount = amount;
+
+        return tokenAmount;
+    }
+
+    public static fromScaledAmount(symbol: string, scaledAmount: BigNumber) {
+        const tokenAmount = new TokenAmount(symbol);
+        const decimals = tokenAmount.decimals;
+        tokenAmount.amount = scaledAmount.div(new BigNumber(10).pow(decimals.toNumber()));
+
+        return tokenAmount;
+    }
+
+    public amount: BigNumber;
+    public readonly token: Token;
+
+    constructor(symbol: string) {
```

Comment:
This constructor should be marked as `private` or `protected` to enforce the use of the static methods.